ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala

```diff
@@ -216,45 +219,47 @@ private[events] object TransactionLogUpdatesConversions {
         executionContext: ExecutionContext,
     ): ToApi[GetTransactionTreesResponse] =
       transaction =>
-        Future
-          .traverse(transaction.events)(
-            toTransactionTreeEvent(requestingParties, verbose, lfValueTranslation)
-          )
-          .map { treeEvents =>
-            val visible = treeEvents.map(_.eventId)
-            val visibleOrder = visible.view.zipWithIndex.toMap
-            val eventsById = treeEvents.iterator
-              .map(e =>
-                e.eventId -> e
-                  .filterChildEventIds(visibleOrder.contains)
-                  // childEventIds need to be returned in the event order in the original transaction.
-                  // Unfortunately, we did not store them ordered in the past so we have to sort it to recover this order.
-                  // The order is determined by the order of the events, which follows the event order of the original transaction.
-                  .sortChildEventIdsBy(visibleOrder)
-              )
-              .toMap
-
-            // All event identifiers that appear as a child of another item in this response
-            val children = eventsById.valuesIterator.flatMap(_.childEventIds).toSet
-
-            // The roots for this request are all visible items
-            // that are not a child of some other visible item
-            val rootEventIds = visible.filterNot(children)
-
-            GetTransactionTreesResponse(
-              Seq(
-                TransactionTree(
-                  transactionId = transaction.transactionId,
-                  commandId = getCommandId(transaction.events, requestingParties),
-                  workflowId = transaction.workflowId,
-                  effectiveAt = Some(timestampToTimestamp(transaction.effectiveAt)),
-                  offset = ApiOffset.toApiString(transaction.offset),
-                  eventsById = eventsById,
-                  rootEventIds = rootEventIds,
+        Future.delegate {
```

Comment:
Future.delegate forces everything in its contain block to run on the provided execution context. This helps in cases where:
* you don't know that a Future-returning method is actually fully async
* you do know that a method that is called is executing some of its work async.

In this case, we do know that these conversions boil down to a sync call to the Engine (for details, I'll just defer to Marton's detailed explanation in [here](https://github.com/digital-asset/daml/pull/13852#discussion_r876939622).

Reply:
Yes, I think this makes some sense. We have a task about revising our futures in the api mapping domain, this is frontrunning that one.
Although, I am thinking too this might be homeopathic ( ;) ) since it is the equivalent of Future {}...or not?